pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -115,6 +126,85 @@ func (i *DBIngestor) parseData(tts []prompb.TimeSeries, req *prompb.WriteRequest
 	return dataSamples, rows, nil
 }
 
+// Parse data into a set of samplesInfo infos per-metric.
+// returns: map[metric name][]SamplesInfo, total rows to insert
+// NOTE: req will be added to our WriteRequest pool in this function, it must
+//       not be used afterwards.
+// When Prometheus & Promscale are running HA mode the below parseData is used
+// to validate leader replica samples & ha_locks in TimescaleDB.
+func (i *DBIngestor) parseHAData(tts []prompb.TimeSeries, req *prompb.WriteRequest, ha *ha.HA) (map[string][]model.SamplesInfo, int, error) {
+	dataSamples := make(map[string][]model.SamplesInfo)
+	rows := 0
+
+	var minT, maxT int64
+
+	s, _, err := model.LabelProtosToLabels(tts[0].Labels)
+	if err != nil {
+		return nil, rows, err
+	}
+
+	replicaName := s.GetReplicaName()
+	clusterName := s.GetClusterName()
+
+	if replicaName == "" && clusterName == "" {
```

Comment:
```suggestion
	if replicaName == "" || clusterName == "" {
```
also, I think we should use constants for " __replica__ " and "__cluster__"